example/example.go

```diff
@@ -94,3 +141,105 @@ func main() {
 		log.Printf("Public ID: %v, URL: %v\n", asset.PublicID, asset.SecureURL)
 	}
 }
+
+// Generate a responsive srcset string for a given public ID by building URLs at multiple widths.
+func generateResponsiveSrcSet(cld *cloudinary.Cloudinary, publicID string) (string, error) {
+	widths := []int{200, 400, 800, 1200}
+	var parts []string
+
+	for _, w := range widths {
+		img, err := cld.Image(publicID)
+		if err != nil {
+			return "", fmt.Errorf("failed to initialize image %s: %w", publicID, err)
+		}
+		img.Transformation = fmt.Sprintf("c_scale,w_%d/f_auto/q_auto", w)
+
+		url, err := img.String()
+		if err != nil {
+			return "", fmt.Errorf("failed to build URL for width %d: %w", w, err)
+		}
+		parts = append(parts, fmt.Sprintf("%s %dw", url, w))
+	}
+
+	return strings.Join(parts, ", "), nil
+}
+
+// Upload a video with transformations applied on upload to generate posters or clips.
+func uploadVideoWithTransformations(cld *cloudinary.Cloudinary, ctx context.Context) *uploader.UploadResult {
+	uploadResult, err := cld.Upload.Upload(
+		ctx,
+		videoFilePath,
+		uploader.UploadParams{
+			PublicID:     "promo_clip",
+			Folder:       "videos/promos",
+			ResourceType: "video",
+			Eager:        "c_fill,h_360,w_640,b_black|c_crop,ar_16:9,e_volume:0.5,du_15",
+			Tags:         []string{"video", "promo"},
+		},
+	)
+	if err != nil {
+		log.Fatalf("failed to upload video: %v", err)
+	}
+
+	// uploadResult.SecureURL points to the original video; uploadResult.Eager to derivatives.
+	return uploadResult
```

Comment:
it's worth to improve error checking here, please check `uploadResult.Error.Message`, there was an error for me.